### PR TITLE
Prefix.pch: Don't import TargetConditionals.h unless neccessary, correctly undefine _THEOS_IS_SIMULATOR

### PR DESCRIPTION
**Pull request by opa334**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Adresses concerns about my previous PR #596 (see comments)

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
No

Where has this been tested?
---------------------------
**Operating System:** macOS

**Platform:** macOS

**Target Platform:** iOS

**Toolchain Version:** Xcode 12.5

**SDK Version:** iPhoneOS13.0

### CODE REVIEW

**Review by PoomSmart — APPROVED** [review]

File: Prefix.pch [hunk 1]
```diff
@@ -10,30 +10,31 @@
...
+			#if (TARGET_OS_IPHONE || _THEOS_IS_SIMULATOR)
+				#import <theos/BackwardsCompat.h>
+				#import <Foundation/Foundation.h>
+				#import <UIKit/UIKit.h>
+				#import <theos/IOSMacros.h>
+				#import <HBLog.h>
+			#endif
 
-		#undef _THEOS_IS_IOS
+			#undef _THEOS_IS_SIMULATOR
```
Comment (kirb) [inline Prefix.pch:37]: Good catch. Indeed this #undef should have been for the simulator constant.

**Review by kirb — CHANGES_REQUESTED** [review]
Thanks heaps for the PR! Just spotted one issue where the `TARGET_OS_IPHONE` block will always be followed because it was changed to `#ifdef`, but `TARGET_OS_IPHONE` will always be defined (to `0` or `1`). Also just got one minor nitpick you could clean up while you’re fixing the other issue.

**Comment by opa334** [comment]
@kirb I just tested something and `TARGET_OS_IPHONE` is 1 even if the package is being compiled for simulator, so I'm pretty sure the `_THEOS_IS_SIMULATOR` check is unnecessary, it only was necessary when theos was still using `TARGET_IPHONE` (on simulator compile `TARGET_OS_IPHONE` is 1 and `TARGET_IPHONE` is not defined at all, on non-simulator compile both are 1), so I have removed the check now entirely (and switched it to #if as requested).

**Comment by kirb** [comment]
Looks good. `TARGET_IPHONE` and `TARGET_IPHONE_SIMULATOR` was defined by Theos once upon a time. Seems I was confused for a while whether this was an actual SDK thing or not. I checked and `TARGET_OS_IPHONE` goes back to at least iOS 4.3 SDK (we support 5.1+) so we’re good to go on that.

**Review by kirb — APPROVED** [review]

**Review by kabiroberai — APPROVED** [review]

File: Prefix.pch [hunk 2]
```diff
@@ -10,30 +10,31 @@
...
+			#endif
 
-		#if (TARGET_OS_IPHONE || _THEOS_IS_SIMULATOR) && !defined(__IPHONE_14_0) && !defined(THEOS_LEAN_AND_MEAN)
-			#import <theos/BackwardsCompat.h>
-			#import <Foundation/Foundation.h>
-			#import <UIKit/UIKit.h>
-			#import <theos/IOSMacros.h>
-			#import <HBLog.h>
-		#endif
+			#if (TARGET_OS_IPHONE || _THEOS_IS_SIMULATOR)
```
[the remark this answers is not in the document]
Comment (kirb) [reply Prefix.pch]: I think I confused the Theos-defined TARGET_IPHONE (which shouldn’t really exist, it’s confusing) and SDK-defined TARGET_OS_IPHONE when writing this. Pretty confident this new check is right. This block should apply to iOS and iOS simulator (TARGET_OS_IPHONE is 1 for all iOS-derived platforms while TARGET_OS_IOS only 1 for true iOS).

**Review by PoomSmart — APPROVED** [review]

**Comment by kirb** [comment]
Merged. Thanks!

**Pull request merged**